src/run.rs

```diff
@@ -303,6 +304,64 @@ fn image_ids_in_use() -> io::Result<HashSet<String>> {
     Ok(image_ids)
 }
 
+// Find docker root directory.
+fn docker_root_dir() -> io::Result<String> {
+    // Query Docker for the root directory.
+    let output = Command::new("docker")
+        .args(&["info", "--format", "{{.DockerRootDir}}"])
+        .stderr(Stdio::inherit())
+        .output()?;
+
+    // Ensure the command succeeded.
+    if !output.status.success() {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            "Unable to determine the Docker root directory.",
+        ));
+    }
+
+    // Find the relevant line of output.
+    String::from_utf8(output.stdout)
+        .map_err(|error| io::Error::new(io::ErrorKind::Other, error))
+        .map(|s| s.trim().to_string())
+}
+
+fn prefix_match_len(s1: &str, s2: &str) -> i32 {
+    let size = std::cmp::min(s1.len(), s2.len());
+    let mut matching_len = 0;
+    for i in 0..size {
+        if s1.chars().nth(i) == s2.chars().nth(i) {
+            matching_len += 1;
+        } else {
+            break;
+        }
+    }
+    return matching_len;
+}
+
+fn get_disk_by_file<'a>(disks: &'a [Disk], path: &str) -> &'a Disk {
+    let mut longest_match_index: usize = 0;
+    let mut longest_match = 0;
+    for (i, disk) in disks.iter().enumerate() {
+        let n = prefix_match_len(disk.mount_point().to_str().unwrap_or(""), path);
+        if n > longest_match {
+            longest_match = n;
+            longest_match_index = i;
+        }
+    }
+
+    &disks[longest_match_index]
+}
+
+// Find size of filesystem on which docker root directory is stored.
+fn docker_root_dir_filesystem_size() -> io::Result<Byte> {
+    let root_dir = docker_root_dir()?;
+    let sys = System::new_all();
```

Comment:
To avoid needing to enumerate all the hardware, can we use `System::new_with_specifics` to only enumerate the disks here? (I see `sys.disks()` below, but I think using `System::new_all` here involves listing all the hardware unnecessarily.)

I also see from the documentation of this crate that we should keep this `System` thing around and reuse it, rather than recreating it every time, for efficiency. Can we do that?